contracts/rewarder/TimeBasedMasterChefRewarder.sol

```diff
@@ -86,8 +98,9 @@ contract TimeBasedMasterChefRewarder is IRewarder, Ownable {
 
         if (pending > 0) {
             rewardToken.safeTransfer(recipient, pending);
+            totalPendingRewards = totalPendingRewards - pending;
```

Comment:
This could underflow since you call `_checkpointPendingRewards()` after. totalPendingRewards could be 0 still.